### PR TITLE
Improve performance when logging is disabled

### PR DESCRIPTION
**Pull request by tomer-w**
As described in: https://github.com/Koenkk/zigbee2mqtt/pull/23893, this PR change the expensive usage of ```JSON.stringify``` usage in tracing to use the new lambda functionality.

### CODE REVIEW

File: src/adapter/z-stack/znp/znp.ts [hunk 1]
```diff
@@ -69,26 +69,22 @@ class Znp extends events.EventEmitter {
...
-            if (type === Type.SREQ) {
-                logger.debug(`SREQ: ${message}`, NS);
-            } else {
-                throw new Error(`Unknown type '${type}'`);
-            }
+    private static stringifyObject(prefix: string, object: ZpiObject): string {
+        return `${Type[object.type]}: ${prefix} ${Subsystem[object.subsystem]} - ${object.command} - ${JSON.stringify(object.payload)}`;
+    }
+
+    private log(prefix: string, object: ZpiObject): void {
```
Comment (Nerivec) [inline src/adapter/z-stack/znp/znp.ts:76]: Should remove `stringifyObject` and `log` functions and declare `toString` in `ZpiObject`

```typescript
    public toString(direction: '<--' | '-->'): string {
        return `${Type[this.type]}: ${direction} ${Subsystem[this.subsystem]} - ${this.command} - ${JSON.stringify(this.payload)}`;
    }
```
then just call `logger.debug(() => object.toString('<--'), NS);` directly in the 2 places where it's used.

File: src/controller/model/endpoint.ts [hunk 2]
```diff
@@ -714,10 +714,10 @@ class Endpoint extends Entity {
             optionsWithDefaults.reservedBits,
         );
 
-        const log =
+        const messageBuilder = (): string =>
             `CommandResponse ${this.deviceIeeeAddress}/${this.ID} ` +
             `${cluster.name}.${command.name}(${JSON.stringify(payload)}, ${JSON.stringify(optionsWithDefaults)})`;
-        logger.debug(log, NS);
+        logger.debug(() => messageBuilder(), NS);
```
Comment (Nerivec) [inline src/controller/model/endpoint.ts:720]: No wrapping anon function necessary here, can just pass `messageBuilder` as param. Same for all other occurrences in the PR.

File: src/controller/model/endpoint.ts [hunk 3]
```diff
@@ -739,7 +739,7 @@ class Endpoint extends Entity {
             });
         } catch (error) {
             const err = error as Error;
-            err.message = `${log} failed (${err.message})`;
+            err.message = `${messageBuilder()} failed (${err.message})`;
             logger.debug((err as Error).stack!, NS);
```
Comment (Nerivec) [inline src/controller/model/endpoint.ts:743]: Should cleanup these (even though not strictly related to this PR, looks like there are a few occurrences).
`(err as Error)` => `err` (already declared as such with `const err = error as Error;`)

File: src/adapter/z-stack/znp/znp.ts (hunk 1 above)
Comment (tomer-w) [reply src/adapter/z-stack/znp/znp.ts:76]: I made the change, but since I don't think direction belongs to the object, I kept it outside of the toString().

File: src/controller/model/endpoint.ts (hunk 2 above)
Comment (tomer-w) [reply src/controller/model/endpoint.ts:720]: Done.

File: src/controller/model/endpoint.ts (hunk 3 above)
Comment (tomer-w) [reply src/controller/model/endpoint.ts:743]: Fixed.

**Comment by tomer-w** [comment]
I submitted all the changes after the CR. Thanks!

File: src/controller/model/endpoint.ts (hunk 2 above)
Comment (Nerivec) [reply src/controller/model/endpoint.ts]: @Koenkk Your opinion on the name for these functions? `messageBuilder` doesn't follow the usual "action/verb" pattern of a function. `getLogString`?

File: src/adapter/z-stack/znp/znp.ts [hunk 4]
```diff
@@ -69,26 +69,10 @@ class Znp extends events.EventEmitter {
...
-            }
-        }
-    }
-
     private onUnpiParsed(frame: UnpiFrame): void {
         try {
             const object = ZpiObject.fromUnpiFrame(frame);
-            const message = `<-- ${Subsystem[object.subsystem]} - ${object.command} - ${JSON.stringify(object.payload)}`;
-            this.log(object.type, message);
+            logger.debug(() => `<-- ${object}`, NS);
```
Comment (Koenkk) [inline src/adapter/z-stack/znp/znp.ts:75]: Why was this changed?

Comment (Koenkk) [reply src/adapter/z-stack/znp/znp.ts:75]: Never mind, just noticed the `toString()`

File: src/controller/model/endpoint.ts (hunk 2 above)
Comment (Koenkk) [reply src/controller/model/endpoint.ts]: What about `createLogMessage`?

**Comment by Koenkk** [comment]
Thanks @tomer-w !
